optuna/storages/_cached_storage.py

```diff
@@ -204,15 +188,13 @@ def set_trial_state(self, trial_id: int, state: TrialState) -> bool:
                 assert cached_trial.state != TrialState.WAITING
 
                 self._check_trial_is_updatable(cached_trial)
-                updates = self._get_updates(trial_id)
-                cached_trial.state = state
-                updates.state = state
+                ret = self._backend.set_trial_state(trial_id, state)
 
+                cached_trial.state = state
                 if cached_trial.state.is_finished():
-                    now = datetime.datetime.now()
-                    updates.datetime_complete = now
-                    cached_trial.datetime_complete = now
-                return self._flush_trial(trial_id)
```

Comment:
📝 Last flush in `Study.tell`, which also in addition to the state used to include the trial value(s). After the change, the value is already "flushed" as this point.